File: b&b/FE/bandb/src/DTO/calendarioDTO.ts
import { prenotazioneDTO } from "./prenotazioneDTO";

export class calendarioDTO{

    id: number;

    mese: number;

    anno: number;

    prenotazione: prenotazioneDTO[];

constructor(id: number,mese : number,anno: number,prenotazione:prenotazioneDTO[]){
    this.id = id;
    this.mese = mese;
    this.anno = anno;
    this.prenotazione = prenotazione;
}
    // private Long id;

    // private int mese;

    // private int anno;

    // private List<Prenotazione> prenotazioni;
}

File: b&b/BE/bandbmichele/src/main/java/it/unicam/dto/CalendarioDTO.java
package it.unicam.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import it.unicam.model.Prenotazione;
import lombok.*;

import java.util.List;

@Data
@NoArgsConstructor
@JsonIdentityInfo(generator= ObjectIdGenerators.IntSequenceGenerator.class,property="@id", scope = CalendarioDTO.class)
public class CalendarioDTO {

    private Long id;

    private int mese;

    private int anno;

    private List<Prenotazione> prenotazioni;
}
